include/libtorrent/peer_class.hpp

```diff
@@ -47,7 +47,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent
 {
-	typedef boost::uint8_t peer_class_t;
+	typedef boost::uint16_t peer_class_t;
```

Comment:
right 2^16 is an arbitrary limit, and most likely the most restrictive one when adding a lot of torrents